core/embed/rust/src/ui/model_tt/component/keyboard/pin.rs

```diff
@@ -18,7 +18,7 @@ use crate::{
 };
 
 pub enum PinKeyboardMsg {
-    Confirmed,
+    Confirmed(Vec<u8, MAX_LENGTH>),
```

Comment:
Since the `pin()` method exists I guess it was supposed to be used to obtain the pin, but haven't figured out how to get a reference to the layout object at the time the result is constructed.

Reply:
This is exactly the usecase the `ObjComponentMsg` trait (hope I remember the name right) is supposed to address.

Reply:
If you omit the `TryInto` impl for the msg type, you should be able to impl the above trait and get a ref to the component for the conversion. Then you can pull the PIN out of it and convert into Obj.

Reply:
If you omit the `TryInto` impl for the msg type, you should be able to impl the above trait and get a ref to the component for the conversion. Then you can pull the PIN out of it and convert into Obj.

Reply:
When I implement the trait like https://github.com/trezor/trezor-firmware/blob/31fc2a4270b01cdbd4d8aeb33b34ba45065c9cb6/core/embed/rust/src/ui/model_tt/layout.rs#L57-L65
then I get:
```
error[E0119]: conflicting implementations of trait `ui::layout::obj::ComponentMsgObj` for type `ui::model_tt::component::keyboard::pin::PinKeyboard`
  --> src/ui/model_tt/layout.rs:57:1
   |
57 |   impl ComponentMsgObj for PinKeyboard
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `ui::model_tt::component::keyboard::pin::PinKeyboard`
   |
  ::: src/ui/layout/obj.rs:37:1
   |
37 | / impl<T> ComponentMsgObj for T
38 | | where
39 | |     T: Component,
40 | |     T::Msg: TryInto<Obj, Error = Error>,
...  |
44 | |     }
45 | | }
   | |_- first implementation here
```
What am I missing?

Reply:
The `TryInto` implementation is not there anymore, perhaps the compiler is not smart enough to figure out that `T::Msg: TryInto<Obj, Error = Error>` is false?

Reply:
As discussed, we may be hitting rust-lang/rust#20400. I rewrote all the `TryInto` conversions to explicit `ComponentMsgObj` impls, it's a bit ugly but uses less magic.